src/model/data_frame.cpp

```diff
@@ -1,2 +1,22 @@
-
 #include "data_frame.h"
+#include "services/service_manager.h"
+#include "utility/json_utils.h"
+#include "utility/time_utils.h"
+#include <boost/property_tree/ptree.hpp>
+
+record_data_frame_t record_data_frame_t::createDataFrame() {
+    const std::vector<std::string>& keys = SM::getDataCache()->getKeys();
+    const boost::property_tree::ptree cache = SM::getDataCache()->toPropertyTree();
```

Comment:
Do you think we need to add a mutex on the cache to make sure that the cache isn't updated between these two calls?

Reply:
Just thinking about if we decide to remove a key from the cache after its value hasn't changed in a while (like you mentioned was done for solar splash). If this happened between these two calls then as we iterate through the keys below the removed key wouldn't be found in the property tree.

Reply:
Actually nvm, the calls to getKeys isn't even needed. I should be able to just iterate over the property tree itself. I'll do that when I move the data point creation code out of here.